src/Mobs/Enderman.cpp

```diff
@@ -194,14 +194,37 @@ void cEnderman::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		return;
 	}
 
-	// TODO take damage in rain
-
 	// Take damage when touching water, drowning damage seems to be most appropriate
-	if (IsSwimming())
+	if (CheckRain() || IsSwimming())
 	{
 		EventLosePlayer();
 		TakeDamage(dtDrowning, nullptr, 1, 0);
 		// TODO teleport to a safe location
 	}
+}
+
+
 
+
+
+bool cEnderman::CheckRain(void)
+{
+	if (GetWorld()->IsWeatherRain())
+	{
+		Vector3d coords = GetPosition();
+		for (int Y = static_cast<int>(coords.y); Y < cChunkDef::Height; ++Y)
+		{
+			BLOCKTYPE Block = m_World->GetBlock(static_cast<int>(coords.x), Y, static_cast<int>(coords.z));
+			if (Block != E_BLOCK_AIR)
```

Comment:
I tested this in vanilla and they don't take damage from rain when they are standing under a fence or another non-solid block.